addon/mixins/disposable.js

```diff
@@ -57,46 +56,6 @@ export default Mixin.create({
   @public
   */
   registerDisposable(dispose) {
-    assert(
-      'Called `registerDisposable` where `dispose` is not a function',
-      typeof dispose === 'function'
-    );
-
-    let disposables = getOrAllocate(this, '_registeredDisposables', Array);
-    let disposable = toDisposable(dispose);
-
-    disposables.push(disposable);
-
-    return disposable;
-  },
-
-  willDestroy() {
```

Comment:
IMHO, we should still implement `willDestroy` in the mixin so we can avoid adding a new (and unique) function to each instance...

Reply:
So you think each mixin should call `runDisposables` themselves?

Reply:
Done. I've added the `willDestroy`s back in and marked the method as patched.